docs/subnets/customize-a-subnet.md

```diff
@@ -211,68 +211,103 @@ If `allowFeeRecipients` feature is enabled on the Subnet, but a validator doesn'
+The `AllowList` solidity interface is defined as follows:
 
 ```solidity
-// (c) 2022-2023, Ava Labs, Inc. All rights reserved.
-// See the file LICENSE for licensing terms.
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
 
-// SPDX-License-Identifier: MIT
+interface IAllowList {
+  // Set [addr] to have the admin role over the precompile
+  function setAdmin(address addr) external;
 
-pragma solidity >=0.8.0;
+  // Set [addr] to be enabled on the precompile contract.
+  function setEnabled(address addr) external;
 
-interface AllowListInterface {
-    // Set [addr] to have the admin role over the allow list
-    function setAdmin(address addr) external;
+  // Set [addr] to have no role the precompile contract.
+  function setNone(address addr) external;
 
-    // Set [addr] to be enabled on the allow list
-    function setEnabled(address addr) external;
+  // Read the status of [addr].
+  function readAllowList(address addr) external view returns (uint256 role);
+}
+```
+
+`readAllowList(addr)` will return a uint256 with a value of 0, 1, or 2, corresponding to the roles `None`, `Enabled`, and `Admin` respectively.
+
+_Note: `AllowList` is not an actual contract but just an interface. It's not callable by itself. This is used by other precompiles. Check other precompile sections to see how this works._
 
-    // Set [addr] to have no role over the allow list
-    function setNone(address addr) external;
+### Restricting Smart Contract Deployers
+
+If you'd like to restrict who has the ability to deploy contracts on your
+subnet, you can provide an `AllowList` configuration in your genesis or upgrade file:
 
-    // Read the status of [addr]
-    function readAllowList(address addr) external view returns (uint256);
+```json
+{
+  "contractDeployerAllowListConfig": {
+    "blockTimestamp": 0,
+    "adminAddresses": ["0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"]
+  }
 }
 ```
 
-If you attempt to add a `Deployer` and you are not an `Admin`, you will see
+In this example, `0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC` is named as the
+`Admin` of the `ContractDeployerAllowList`. This enables it to add other `Admin` or to add
+`Enabled` addresses. Both `Admin` and `Enabled` can deploy contracts. To provide
+a great UX with factory contracts, the `tx.Origin` is checked for being a valid
+deployer instead of the caller of `CREATE`. This means that factory contracts will still be
+able to create new contracts as long as the sender of the original transaction is an allow
+listed deployer.
+
+The `Stateful Precompile` contract powering the `ContractDeployerAllowList` adheres to the [AllowList Solidity interface](#allowlist-interface) at `0x0200000000000000000000000000000000000000` (you can load this interface and interact directly in Remix):
+
+If you attempt to add a `Enabled` and you are not an `Admin`, you will see
```

Comment:
```suggestion
- If you attempt to add a `Enabled` and you are not an `Admin`, you will see
```
Should this `If` and the other 2 ifs be under this `:`?

Reply:
This should be the IAllowList here: https://github.com/ava-labs/subnet-evm/blob/master/contract-examples/contracts/IAllowList.sol

Reply:
Not sure I understand what needs to be done with this.

Reply:
We should copy and paste this interface so that it goes under the section following `... directly in Remix):` and shows the user exactly the interface that they can use in Remix.

There is another example of what this looks like for the solidity interfaces below

Reply:
We just had that interface in line 234-251 and we should not duplicate it here.